packages/thrift-client/src/main/connections/index.ts

```diff
@@ -56,7 +55,7 @@ export function createHttpClient<TClient extends ThriftClient<RequestOptions>>(
     }
 
     const connection: HttpConnection = new HttpConnection(
-        deepMerge(options, { serviceName }),
+        Object.assign(options, { serviceName }),
```

Comment:
Even if deep/recursive merges aren't needed, I think this still has an important difference from the previous implementation - this will modify `options`, whereas it looks like `deepMerge` didn't.
```node
> const options = { a: 1, b: 2 }
undefined
> Object.assign(options, { c: 3 })
{ a: 1, b: 2, c: 3 }
> options
{ a: 1, b: 2, c: 3 }
```
I think either
```suggestion
        Object.assign({}, options, { serviceName }),
```
or this would do the trick
```suggestion
        { ...options, serviceName },
```

Reply:
Ooh, nice catch - I'll probably just switch it to the second one, thanks!